superbench/benchmarks/model_benchmarks/pytorch_cnn.py

```diff
@@ -0,0 +1,147 @@
+                    self._name, precision, str(e)
+                )
+            )
+            return False
+
+        self._target = torch.LongTensor(self._args.batch_size).random_(self._args.num_classes)
+        if self._gpu_available:
+            self._target = self._target.cuda()
+
+        return True
+
+    def _train_step(self, precision):
+        """Define the training process.
+
+        Args:
+            precision (Precision): precision of model and input data, such as float32, float16.
+
+        Return:
+            The step-time list of every training step.
+        """
+        duration = []
+        curr_step = 0
+        while True:
+            for idx, sample in enumerate(self._dataloader):
+                start = time.time()
+                sample = sample.to(dtype=getattr(torch, precision.value))
+                if self._gpu_available:
+                    sample = sample.cuda()
+                self._optimizer.zero_grad()
+                output = self._model(sample)
+                loss = self._loss_fn(output, self._target)
+                loss.backward()
+                self._optimizer.step()
+                end = time.time()
+                curr_step += 1
+                if curr_step > self._args.num_warmup:
+                    # Save the step time of every training/inference step, unit is millisecond.
+                    duration.append((end - start) * 1000)
+                if self._is_finished(curr_step, end):
+                    return duration
+
+    def _inference_step(self, precision):
+        """Define the inference process.
+
+        Args:
+            precision (Precision): precision of model and input data,
+              such as float32, float16.
+
+        Return:
+            The latency list of every inference operation.
+        """
+        duration = []
+        curr_step = 0
+        with torch.no_grad():
+            self._model.eval()
+            while True:
+                for idx, sample in enumerate(self._dataloader):
+                    start = time.time()
+                    sample = getattr(sample, precision.value)
+                    if self._gpu_available:
+                        sample = sample.cuda()
+                    self._model(sample)
+                    if self._gpu_available:
+                        torch.cuda.synchronize()
+                    end = time.time()
+                    curr_step += 1
+                    if curr_step > self._args.num_warmup:
+                        # Save the step time of every training/inference step, unit is millisecond.
+                        duration.append((end - start) * 1000)
+                    if self._is_finished(curr_step, end):
+                        return duration
+
+
+# Register CNN benchmarks.
+# Reference: https://pytorch.org/vision/stable/models.html
+for model in ['resnet50', 'resnet101', 'resnet152', 'densenet169', 'densenet201', 'vgg11', 'vgg13', 'vgg16', 'vgg19']:
```

Comment:
Maybe we can add more for this?

Reply:
Do you think we could add all the cnn models?

Reply:
Yes, although user may not use it in most of time ~